### PR TITLE
LPS-191268 Change status to SCHEDULED if displayDate is on the future

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
Adding the back functionallity to check if the display date is on the future and put the status

As we can not check it with the portal till the front is made, and not changing manually UpdateKBArticleMVCActionCommand I added some test to check the new behavior

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:knowledge-base

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by austinchiang** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java [hunk 1]
```diff
@@ -2340,16 +2349,19 @@ private void _validate(
 			throw new KBArticleDisplayDateException("Display date is null");
 		}
 
-		_validateExpirationReviewDate(expirationDate, reviewDate);
+		_validateExpirationReviewDate(displayDate, expirationDate, reviewDate);
 	}
 
 	private void _validateExpirationReviewDate(
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java:2355]: Shouldn't we rename this method to include the display date?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by austinchiang** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are unrelated.

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**